### ISSUE TITLE
Search field position is not updated after search cancellation

### ISSUE THREAD

**Issue by samy** [select2/select2#4936]
## Prerequisites
-  Check Select2 examples page

## Steps to reproduce the issue

1. Go on https://select2.github.io/examples.html#data-ajax
2. Scroll to put "select" element at bottom of the page (the results will be displayed on top of the select)
3. Make some search
4. Cancel search by pressing Esc
5. Click a second time on select field

## Expected behavior and actual behavior

When I follow those steps, I see...
![after_search](https://cloud.githubusercontent.com/assets/1282106/26728417/756a6a26-47aa-11e7-870c-879f24c0be21.png)

I was expecting...
![sticky_search](https://cloud.githubusercontent.com/assets/1282106/26728425/7a894acc-47aa-11e7-9038-4abddafb8034.png)

## Environment

Browsers

- [x] Google Chrome
- [ ] Mozilla Firefox
- [ ] Internet Explorer

Operating System

- [x] Windows
- [ ] Mac OS X
- [ ] Linux
- [ ] Mobile

Libraries

- jQuery version: 1.11.2
- Select2 version: 4.0.3

## Isolating the problem

- It appears only after search cancelling

**Comment by alexweissman** [comment]
Duplicate of #4614.  Please search the issues carefully before opening a new issue!
